### PR TITLE
Refactoring to Redux

### PR DESCRIPTION
**Pull request by b0llu**
What to check,
- [ ] src/Redux/Reducers/authSlice.js
- [ ] src/Redux/Reducers/postsSlice.js
- [ ] src/Redux/Reducers/usersSlice.js
- [ ] src/Redux/store.js

What not to check,
- [ ] There are multiple changes in multiple files but only check redux related once 

live Preview,
https://stormmedia.netlify.app/

### CODE REVIEW

File: src/Redux/Reducers/authSlice.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+const encodedToken = localStorage.getItem("StormMediaToken");
+const initialState = {
+  currentUser: {},
+};
+
+export const login = createAsyncThunk("auth/login", async (userDetails) => {
+  try {
+    const response = await axios.post(`/api/auth/login`, {
+      username: userDetails.username,
+      password: userDetails.password,
```
Comment (govindappaarun) [inline src/Redux/Reducers/authSlice.js:13]: Suggestion: if you spread user details, you can avoid creating an object again

File: src/Redux/Reducers/authSlice.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  } catch (error) {
+    AlertToast(`${error.response.data.errors}`);
+  }
+});
+
+export const signup = createAsyncThunk("auth/signup", async (userDetails) => {
+  try {
+    const response = await axios.post(`/api/auth/signup`, {
+      firstName: userDetails.name,
+      username: userDetails.username,
```
Comment (govindappaarun) [inline src/Redux/Reducers/authSlice.js:25]: Suggestion: same comments as for login, and try to match name of keys similar to backend ( passwordOne as just password ) so mapping can be avoided

**Review by govindappaarun — APPROVED** [review]
Redux part looks good to merge

File: src/Redux/Reducers/authSlice.js (hunk 1 above)
Comment (b0llu) [reply src/Redux/Reducers/authSlice.js:13]: Thanks for suggestion will look into it

File: src/Redux/Reducers/authSlice.js (hunk 2 above)
Comment (b0llu) [reply src/Redux/Reducers/authSlice.js:25]: Thanks for the suggestion will look into it

**Review by aayushjain-code — APPROVED** [review]
Very Nicely coded every piece.
The addition of Redux is a great plus keep progressing

**Pull request merged**